src/platform_impl/web/device.rs

```diff
@@ -1,8 +1,26 @@
+use web_sys::PointerEvent;
+
 #[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct DeviceId(pub i32);
+pub struct DeviceId {
```

Comment:
This semantically changes `DeviceId` from just being an "identifier", to being a "container", i.e. it should now probably be named `DeviceData`.

The change I'd rather like to see is to change it to a "handle", as I believe it would be useful for e.g. the gamepad work or querying other device information. Ideally, I think that should be done in a separate PR first (i.e. a PR to rename `DeviceId` to `DeviceHandle`, and perhaps store pointers instead of just a `usize` derived from the pointer). Such a change also affects https://github.com/rust-windowing/winit/pull/3784, as now we suddenly have to do extra work to handle possibly-invalid device identifiers (whereas before it was just UB).

Reply:
Unfortunately I'm clueless about how exactly this works on other platforms, on Web the device ID has no data associated with it and can't be used for anything else then comparing it to other device IDs. So a device handle would have use on Web.

The `primary` data I'm adding here is actually associated with the event and not with the device ID, but I can't really add it to the event without cross-platform support.

Fortunately the device ID on Web is not used outside those events so it all makes sense.

Reply:
Solved by going with a completely different route introducing a new type `FingerId`.